eb_hooks.py

```diff
@@ -450,6 +450,16 @@ def pre_fetch_hook_zen4_gcccore1220(self, *args, **kwargs):
         print_msg("Updated build option 'force' to 'True'")
 
 
+def pre_install_hook(self, *args, **kwargs):
+    """Main pre install hook: trigger custom functions based on software name."""
+    if 'keeppreviousinstall' in self.cfg and self.cfg['keeppreviousinstall']:
```

Comment:
@bedroge I think we deserve some context here, where/how we're using `keeppreviousinstall`, and why we only want to use this only when empty directories are available.
Definitely point to https://github.com/EESSI/software-layer/issues/556 + https://github.com/EESSI/software-layer/pull/871

Reply:
Done: https://github.com/EESSI/software-layer/pull/881/commits/cc79fd7f1ab09e6857f4c894f1d2c69ded67e365